java/src/com/ibm/streamsx/topology/kafka/KafkaMessage.java

```diff
@@ -0,0 +1,28 @@
+package com.ibm.streamsx.topology.kafka;
+
+import java.io.Serializable;
+
+/**
+ * The components of an Apache Kafka Message.
+ */
+public interface KafkaMessage extends Serializable {
```

Comment:
A different approach, would be to support Kafka messages as JSONObject, which has strong existing interchange capabilities with SPL. Would neatly support the optional key and topic.